airbyte-webapp/src/views/Connector/ConnectorForm/components/Sections/ArraySection.tsx

```diff
@@ -79,10 +103,13 @@ export const ArraySection: React.FC<ArraySectionProps> = ({ formField, path, dis
           render={(arrayHelpers) => (
             <ArrayOfObjectsEditor
               editableItemIndex={editIndex}
-              onStartEdit={setEditIndex}
+              onStartEdit={(n) => {
+                setEditIndex(n);
+                setItemInEdit(items[n]);
```

Comment:
Do we not need to create a copy of this?

Reply:
yes, to avoid the live changes in the modal form to be reflected behind the backdrop as well as to restore the old state in case the user cancels.